src/main/java/org/sonatype/repository/vgo/internal/hosted/HostedHandlers.java

```diff
@@ -0,0 +1,109 @@
+
+import static com.google.common.base.Preconditions.checkNotNull;
+import static org.sonatype.nexus.repository.http.HttpResponses.notFound;
+import static org.sonatype.nexus.repository.http.HttpResponses.ok;
+
+@Named
+@Singleton
+public class HostedHandlers
+    extends ComponentSupport
+{
+  private VgoPathUtils pathUtils;
+
+  @Inject
+  public HostedHandlers(final VgoPathUtils pathUtils) { this.pathUtils = checkNotNull(pathUtils); }
+
+  final Handler get = context -> {
+    VgoAssetKind assetKind = context.getAttributes().require(VgoAssetKind.class);
+    Content content = null;
+    switch (assetKind) {
+      case VGO_INFO:
+        content = getInfo(context);
+        break;
+      case VGO_MODULE:
+        content = getModule(context);
+        break;
+      case VGO_PACKAGE:
+        content = getPackage(context);
+        break;
+      case VGO_LIST:
+        content = getList(context);
+        break;
+    }
+
+    return (content != null) ? ok(content) : notFound();
+  };
+
+  private Content getList(final Context context) {
+    Content content;
+    State state = context.getAttributes().require(State.class);
+    String module = pathUtils.module(state);
+    content = context.getRepository().facet(VgoHostedFacet.class).getList(module);
+    return content;
+  }
+
+  private Content getPackage(final Context context) {
+    Content content;
+    State state = context.getAttributes().require(State.class);
+    String path = pathUtils.assetPath(state);
+    VgoAttributes vgoAttributes = pathUtils.getAttributesFromMatcherState(state);
+    content = context.getRepository().facet(VgoHostedFacet.class).getPackage(path);
+    return content;
+  }
+
+  private Content getModule(final Context context) {
+    Content content;
+    State state = context.getAttributes().require(State.class);
+    String path = pathUtils.assetPath(state);
+    VgoAttributes vgoAttributes = pathUtils.getAttributesFromMatcherState(state);
+    content = context.getRepository().facet(VgoHostedFacet.class).getMod(path);
+    return content;
+  }
+
+  private Content getInfo(final Context context) {
+    Content content;
+    State state = context.getAttributes().require(State.class);
+    String path = pathUtils.assetPath(state);
+    VgoAttributes vgoAttributes = pathUtils.getAttributesFromMatcherState(state);
+    content = context.getRepository().facet(VgoHostedFacet.class).getInfo(path, vgoAttributes);
+    return content;
+  }
+
+  final Handler upload = context -> {
+    State state = context.getAttributes().require(TokenMatcher.State.class);
+    String path = pathUtils.assetPath(state);
+    VgoAttributes vgoAttributes = pathUtils.getAttributesFromMatcherState(state);
+
+    VgoAssetKind assetKind = context.getAttributes().require(VgoAssetKind.class);
+    context.getRepository().facet(VgoHostedFacet.class).upload(path, vgoAttributes, context.getRequest().getPayload(), assetKind);
+
+    return ok();
```

Comment:
Question: should this be created() ?

Reply:
https://github.com/sonatype-nexus-community/nexus-repository-vgo/pull/5/commits/63e83e1927442581512caac3a78bf6efb5c045ed